### PR TITLE
Chess quality of life update

### PR DESCRIPTION
**Pull request by Sindorman**
- You can now fast fill chess board by clicking on it with chess pieces box.

- You can now shake off all pieces off board by clicking on board in your active hand.

- Chess box now picks up all chess pieces of same color as box off the turf by clicking on the chess piece

### CODE REVIEW

**Comment by Sindorman** [comment]
!review

File: code/modules/games/boardgame.dm [hunk 1]
```diff
@@ -239,6 +256,23 @@ obj/item/weapon/board/attackby(obj/item/I as obj, mob/user as mob)
...
+		if(T)
+			for(var/obj/item/weapon/checker/c in T.contents)
+				if(istype(I, /obj/item/weapon/storage/box/chess/red))
+					if(c.piece_color == "red")
+						c.forceMove(b)
+					else
+						continue
+				else if(c.piece_color != "red")
+					c.forceMove(b)
+			to_chat(user, "<span class='notice'>You put all checker pieces into [b].</span>")
```
Comment (BurgerLUA) [inline code/modules/games/boardgame.dm]: put a \the before the [b]

Comment (skull132) [reply code/modules/games/boardgame.dm]: You don't need `\the` if you use an object. It's automatically preceded by an article if the object's name is set up properly. Which it usually is.

File: code/modules/games/boardgame.dm [hunk 2]
```diff
@@ -30,8 +30,25 @@
 	else
 		src.examine(M)
 
-obj/item/weapon/board/attackby(obj/item/I as obj, mob/user as mob)
-	if(!addPiece(I,user))
+/obj/item/weapon/board/attack_self(mob/user)
+	var/choice = alert("Do you want to throw everything off the [src]",,"No","Yes")
```
[the remark this answers is not in the document]
Comment (skull132) [reply code/modules/games/boardgame.dm]: The double comma is intentional, though bad form. It's a `null` argument.

Comment (skull132) [inline code/modules/games/boardgame.dm]:
```suggestion
	var/choice = alert("Do you want to throw everything off the [src]?", null, "No", "Yes")
```

File: code/modules/games/boardgame.dm [hunk 3]
```diff
@@ -30,8 +30,25 @@
...
+	if(choice == "Yes")
+		for(var/obj/item/weapon/checker/c in src.contents)
+			c.forceMove(get_turf(src.loc))
+			num = 0
```
Comment (skull132) [inline code/modules/games/boardgame.dm]:
```suggestion
		num = 0
```

**Comment by Sindorman** [comment]
idk why bot didn't change labels

**Comment by Arrow768** [comment]
Because skull still has a review that is requesting changes

**Pull request merged**